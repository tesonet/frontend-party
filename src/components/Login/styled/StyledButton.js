import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Button } from 'reactstrap';

const StyledContainer = styled(Button)`
  background-color: ${props => props.disabled ? props.theme.btnBgColorDisabled : props.theme.btnBgColorPrimary};
  border: 0;
  font-size: 16px;
  font-weight: 700;
  height: 3rem;
  line-height: 1.875;

  &:hover,
  &:focus,
  &:not(:disabled):not(.disabled):active {
    background-color: ${props => props.theme.btnBgColorHover};
    border: 0;
  }
`;

const StyledButton = props => (
  <StyledContainer color="primary" size="lg" block {...props}>{props.children}</StyledContainer>
);

StyledButton.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
};

export default StyledButton;
