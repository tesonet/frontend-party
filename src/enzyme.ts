import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import fetchMock from "jest-fetch-mock";

declare let global: { fetch: {} };

global.fetch = fetchMock;
configure({ adapter: new Adapter() });
