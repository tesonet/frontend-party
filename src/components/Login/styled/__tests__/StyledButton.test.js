import React from 'react';
import { shallow } from 'enzyme';
import { Button } from 'reactstrap';

import StyledButton from '../StyledButton';

describe('StyledButton component', () => {
  it('should render the <Button /> element.', () => {
    const renderedComponent = shallow(<StyledButton />).dive();

    expect(renderedComponent.type()).toEqual(Button);
    expect(renderedComponent).toMatchSnapshot();
  });
});
