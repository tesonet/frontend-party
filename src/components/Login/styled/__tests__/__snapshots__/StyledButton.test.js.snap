// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledButton component should render the <Button /> element. 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Styled(Button)
    block={true}
    color="primary"
    size="lg"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "block": true,
      "children": undefined,
      "className": "sc-bdVaJa eMjQhF",
      "color": "primary",
      "size": "lg",
      "tag": "button",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "block": true,
        "children": undefined,
        "className": "sc-bdVaJa eMjQhF",
        "color": "primary",
        "size": "lg",
        "tag": "button",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
